doc/vim-go.txt

```diff
@@ -1418,13 +1418,28 @@ explicitly overrides this behavior. For standard Vim behavior, set it to 10.
                                                             *'g:go_list_type'*
 
 Specifies the type of list to use for command outputs (such as errors from
-builds, results from static analysis commands, etc...). The default value
+builds, results from static analysis commands, etc...) when a value is not
+specified for the command in g:go_list_type_commands. The default value
 (empty) will use the appropriate kind of list for the command that was called.
 Supported values are "", "quickfix", and "locationlist". >
 
   let g:go_list_type = ""
 <
 
+                                                   *'g:go_list_type_commands'*
+
+Specifies the type of list to use for command outputs (such as errors from
+builds, results from static analysis commands, etc...). The default value
+(an empty dictionary) will use the appropriate kind of list for the command
+that was called. Supported keys are "GoBuild", "GoErrCheck", "GoFmt",
```

Comment:
I think that "The default value (an empty dictionary) will use the appropriate kind of list for the command that was called." isn't 100% true, as `g:go_list_type` can override it.

Perhaps it's an idea to add a single help entry for both `g:go_list_type` and `g:go_list_type_commands` which describes how vim-go figures out which list type to use? For example something along the lines of:

```
                                                            *'g:go_list_type'* 
                                                   *'g:go_list_type_commands'*

Specifies the type of list to use for command outputs (such as errors from
builds, results from static analysis commands, etc.)
With `g:go_list_type` you can specify which list to open by default. This
default value can be overridden for specific commands with
`g:go_list_type_commands`, which is a dictionary to map a command to a value
("quickfix" or "locationlist").

Supported keys are "GoBuild", "GoErrCheck", "GoFmt", "GoInstall", "GoLint",
"GoMetaLinter", "GoModifyTags" (used for both :GoAddTags and :GoRemoveTags),
"GoRename", "GoRun", and "GoTest".

If neither options are set then vim-go will use the most appropriate kind of
list for the command that was called.

For example to use the location list by default but the quickfix list for
|:GoInstall|: >
  let g:go_list_type = "quickfix"
  let g:go_list_type_commands = {"GoBuild": "locationlist"}
<
```

Reply:
That's possible, but note that the documentation for `g:go_list_type` does document that it is used when `g:go_list_type_commands` does not contain an entry for the command:
> Specifies the type of list to use for command outputs (such as errors from
builds, results from static analysis commands, etc...) when a value is not
specified for the command in g:go_list_type_commands. The default value
(empty) will use the appropriate kind of list for the command that was called.

Reply:
Yeah, but if you use `:help 'g:go_list_type_commands'` then you won't see that part, because it's (probably) off-screen.

Your last commit looks good :+1: